java/code/src/com/redhat/rhn/taskomatic/task/RepoSyncTask.java

```diff
@@ -62,53 +63,36 @@ public String getConfigNamespace() {
     public void execute(JobExecutionContext context) {
         List<Long> channelIds = getChannelIds(context.getJobDetail().getJobDataMap());
 
-        String[] lparams = {"no-errata", "latest", "sync-kickstart", "fail"};
+        List<String> lparams = Arrays.asList("no-errata", "latest", "sync-kickstart", "fail");
         List<String> ltrue = Arrays.asList("true", "1");
-        List<String> params = new ArrayList<>();
-        for (String p : lparams) {
-            if (context.getJobDetail().getJobDataMap().containsKey(p)) {
-                if (ltrue.contains(context.getJobDetail().getJobDataMap().get(p).toString()
-                        .toLowerCase().trim())) {
-                    params.add("--" + p);
-                }
-            }
-        }
+        List<String> params = lparams.stream()
+                .filter(p -> context.getJobDetail().getJobDataMap().containsKey(p) &&
+                    ltrue.contains(context.getJobDetail().getJobDataMap().get(p).toString().toLowerCase().trim()))
+                .map(p -> "--" + p)
+                .collect(Collectors.toList());
 
         for (Long channelId : channelIds) {
             Channel channel = ChannelFactory.lookupById(channelId);
             if (channel != null) {
                 log.info("Syncing repos for channel: {}", channel.getName());
 
                 try {
-                    executeExtCmd(getSyncCommand(channel, params).toArray(new String[0]));
+                    executeExtCmd(getSyncCommand(channel, params).toArray(new String[0]), true);
                 }
                 catch (JobExecutionException e) {
                     NotificationMessage notificationMessage = UserNotificationFactory.createNotificationMessage(
                             new ChannelSyncFailed(channel.getId(), channel.getName(), e.getMessage())
                     );
-                    if (channel.getOrg() == null) {
-                        UserNotificationFactory.storeNotificationMessageFor(notificationMessage,
-                                Collections.singleton(RoleFactory.CHANNEL_ADMIN), Optional.empty());
-                    }
-                    else {
-                        UserNotificationFactory.storeNotificationMessageFor(notificationMessage,
-                                Collections.singleton(RoleFactory.CHANNEL_ADMIN), Optional.of(channel.getOrg()));
-                    }
-
+                    UserNotificationFactory.storeNotificationMessageFor(notificationMessage,
+                            Collections.singleton(RoleFactory.CHANNEL_ADMIN), Optional.ofNullable(channel.getOrg()));
 
                     log.error(e.getMessage());
```

Comment:
Not directly related to your change I know, but shouldn't we continue to next channel after catching the exception?

It seems to me that in case of error we are storing both `ChannelSyncFailed`  and `ChannelSyncFinished` notifications

Reply:
Makes sense to me. The cleaner solution would be to move the success notifications within the `try` block.